graph/server/server.go

```
@@ -6,55 +6,43 @@ import (
 	"html/template"
 	"net/http"
 	"strings"
+	"time"
 
 	"github.com/aporeto-inc/trireme-statistics/influxdb"
 	"github.com/influxdata/influxdb/client/v2"
 )
 
-// GraphData is the struct that holds the json format required for graph to generate nodes and link
-type GraphData struct {
-	Nodes []Nodes `json:"nodes"`
-	Links []Links `json:"links"`
-}
-
-// Nodes which holds pu information
-type Nodes struct {
-	ID        string `json:"id"`
-	Name      string `json:"name"`
-	IPAddress string `json:"ipaddress"`
-}
+// NewGraph is the handler for graph generators
+func NewGraph(httpClient *influxdb.Influxdb, dbname string) *Graph {
 
-// Links which holds the links between pu's
-type Links struct {
-	Source int    `json:"source"`
-	Target int    `json:"target"`
-	Action string `json:"action"`
+	return &Graph{
+		httpClient: httpClient,
+		dbname:     dbname,
+	}
 }
 
 // GetData is called by the client which generates json with a logic that defines the nodes and links
-func GetData(httpClient *influxdb.Influxdb, dbname string) http.HandlerFunc {
+func (g *Graph) GetData(w http.ResponseWriter, r *http.Request) {
 
-	return func(w http.ResponseWriter, r *http.Request) {
-
-		res, err := getContainerEvents(httpClient, dbname)
-		if err != nil {
-			http.Error(w, err.Error(), 0)
-		}
+	err := json.NewEncoder(w).Encode(g.jsonData)
+	if err != nil {
+		http.Error(w, err.Error(), 3)
+	}
+}
 
-		jsonData, err := transform(res, httpClient, dbname)
-		if err != nil {
-			http.Error(w, err.Error(), 2)
-		}
+// Start is used to start generating jsonData for every 15 seconds
+func (g *Graph) Start(interval int) {
 
-		err = json.NewEncoder(w).Encode(jsonData)
-		if err != nil {
-			http.Error(w, err.Error(), 3)
+	go func() {
+		for range time.Tick(time.Second * time.Duration(interval)) {
+			res, _ := g.getContainerEvents()
```

don't ignore the errors, log them.